src/app/src/pages/eval/components/EvalOutputPromptDialog.tsx

```diff
@@ -145,19 +148,147 @@ function CodeDisplay({
+  filters?: {
+    /** Add a filter to the current filter set */
+    add?: (filter: FilterConfig) => void;
+    /** Clear all active filters */
+    reset?: () => void;
+  };
+
+  /**
+   * API operations for evaluation and debugging.
+   */
+  api?: {
+    /** Replay an evaluation with modified parameters */
+    replay?: (params: ReplayEvaluationParams) => Promise<ReplayEvaluationResult>;
+    /** Fetch trace data for debugging */
+    fetchTraces?: (evaluationId: string, signal: AbortSignal) => Promise<Trace[]>;
+  };
+
+  /**
+   * Cloud platform configuration (null if running in OSS mode).
+   */
+  cloudConfig?: CloudConfig | null;
+}
+
+/**
+ * Props for the EvalOutputPromptDialog component.
+ *
+ * This dialog displays detailed information about an evaluation result,
+ * including the prompt, output, grading results, metadata, and traces.
+ *
+ * ## Dependency Injection Pattern
+ *
+ * All external dependencies (store access, API calls, cloud config) are injected
+ * via the `dependencies` prop to enable complete isolation and testing. The component
+ * gracefully handles missing dependencies by disabling features when they're undefined.
+ *
+ * @example
+ * ```tsx
+ * // Basic usage without dependencies (view-only mode)
+ * <EvalOutputPromptDialog
+ *   open={true}
+ *   onClose={handleClose}
+ *   prompt="What is 2+2?"
+ *   output="4"
+ * />
+ *
+ * // Full-featured usage with all dependencies
+ * <EvalOutputPromptDialog
+ *   open={true}
+ *   onClose={handleClose}
+ *   prompt="What is 2+2?"
+ *   output="4"
+ *   dependencies={{
+ *     filters: {
+ *       add: addFilter,
+ *       reset: resetFilters,
+ *     },
+ *     api: {
+ *       replay: handleReplay,
+ *       fetchTraces: handleFetchTraces,
+ *     },
+ *     cloudConfig: { appUrl: 'https://cloud.example.com', isEnabled: true },
+ *   }}
+ * />
+ * ```
+ */
 interface EvalOutputPromptDialogProps {
+  /** Controls dialog visibility */
   open: boolean;
+  /** Callback fired when dialog should close */
   onClose: () => void;
+  /** The prompt text that was evaluated */
   prompt: string;
+  /** Provider/model identifier (displayed in header) */
   provider?: string;
+  /** The output/response from the evaluation */
   output?: string;
+  /** Grading results for evaluation assertions */
   gradingResults?: GradingResult[];
+  /** Additional metadata about the evaluation */
   metadata?: Record<string, any>;
+  /** Evaluation ID for fetching traces and replays */
   evaluationId?: string;
+  /** Test case ID for trace correlation */
   testCaseId?: string;
+  /** Test case index in the evaluation */
   testIndex?: number;
+  /** Prompt index in the evaluation */
   promptIndex?: number;
+  /** Variables used in the prompt template */
   variables?: Record<string, any>;
+  /**
+   * External dependencies for filters, API calls, and cloud config.
+   * All dependencies are optional - features gracefully degrade when missing.
+   */
+  dependencies?: DialogDependencies;
```

Comment:
I'll also note that this is a react anti-pattern for a number of reasons- performance, composibility, readability, etc. I'd replace them with either the dependencies directly, or event handlers / functions that would do whatever action the dependencies are for